src/AxisSets.jl

```diff
@@ -70,5 +70,6 @@ include("flatten.jl")
 include("patterns.jl")
 include("dataset.jl")
 include("indexing.jl")
+include("functions.jl")
```

Comment:
"functions" seems like an awfully generic name for this file. Would "mapping" be better?

Reply:
Well it also includes `merge` and `rekey`. We'll also likely want to add `cat` there as well.

Reply:
Alternatively, I could just move these functions into `dataset.jl`, that file will just be kinda long.

Reply:
I suppose it's fine as-is. We might want to split it up into `mapping.jl` and `concatenation.jl` (or something) later if the file starts getting too long.